### PR TITLE
✨ Finish updating dependency on kcp v0.11.0

### LINKED ISSUE #164: Bump dependency of kcp to its next release
For some historical reason edge-mc currently depends on [this specific commit](https://github.com/kcp-dev/kcp/tree/4506fdc064060b3fe82e1082533f9798b36ba7a5) of kcp, which is not a tagged version. We may want to bump our dependency of kcp to its next release.

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR continues the updates of our dependency on kcp v0.11.0.

Tested with mailbox-controller.
```
I0301 17:31:18.625159   87502 main.go:102] "Command line flag" add_dir_header="false"
I0301 17:31:18.625299   87502 main.go:102] "Command line flag" alsologtostderr="false"
I0301 17:31:18.625307   87502 main.go:102] "Command line flag" concurrency="4"
I0301 17:31:18.625316   87502 main.go:102] "Command line flag" inventory-context="root"
I0301 17:31:18.625321   87502 main.go:102] "Command line flag" inventory-kubeconfig=""
I0301 17:31:18.625328   87502 main.go:102] "Command line flag" log_backtrace_at=":0"
I0301 17:31:18.625333   87502 main.go:102] "Command line flag" log_dir=""
I0301 17:31:18.625337   87502 main.go:102] "Command line flag" log_file=""
I0301 17:31:18.625342   87502 main.go:102] "Command line flag" log_file_max_size="1800"
I0301 17:31:18.625347   87502 main.go:102] "Command line flag" logtostderr="true"
I0301 17:31:18.625352   87502 main.go:102] "Command line flag" one_output="false"
I0301 17:31:18.625359   87502 main.go:102] "Command line flag" server-bind-address=":10203"
I0301 17:31:18.625364   87502 main.go:102] "Command line flag" skip_headers="false"
I0301 17:31:18.625369   87502 main.go:102] "Command line flag" skip_log_headers="false"
I0301 17:31:18.625374   87502 main.go:102] "Command line flag" stderrthreshold="2"
I0301 17:31:18.625379   87502 main.go:102] "Command line flag" v="2"
I0301 17:31:18.625384   87502 main.go:102] "Command line flag" vmodule=""
I0301 17:31:18.625388   87502 main.go:102] "Command line flag" workload-context=""
I0301 17:31:18.625393   87502 main.go:102] "Command line flag" workload-kubeconfig=""
I0301 17:31:18.644243   87502 main.go:237] "Found APIExport view" exportName="workload.kcp.io" serverURL="https://192.168.1.54:6443/services/apiexport/root/workload.kcp.io"
I0301 17:31:18.645314   87502 shared_informer.go:282] Waiting for caches to sync for mailbox-controller
I0301 17:31:18.746454   87502 shared_informer.go:289] Caches are synced for mailbox-controller
I0301 17:31:18.746554   87502 main.go:200] "Informers synced"
I0301 17:31:54.988604   87502 main.go:361] "Created missing workspace" worker=3 wsName="wmtp3o8lb1n1c5uj-w-stest1"
I0301 17:32:40.182098   87502 main.go:361] "Created missing workspace" worker=1 wsName="wmtp3o8lb1n1c5uj-w-stest1"
I0301 17:33:27.751126   87502 main.go:342] "Deleted unwanted workspace" worker=2 wsName="wmtp3o8lb1n1c5uj-w-stest1"
```

Tested with edge-scheduler.
```
I0301 17:27:28.930126   87207 scheduler.go:211] "Found APIExport view" exportName="edge.kcp.io" serverURL="https://192.168.1.54:6443/services/apiexport/wmtp3o8lb1n1c5uj/edge.kcp.io"
I0301 17:27:28.931079   87207 reflector.go:271] Starting reflector *v1alpha1.Location (10h0m0s) from k8s.io/client-go@v0.0.0-20230210192259-aaa28aa88b2d/tools/cache/reflector.go:215
I0301 17:27:28.931144   87207 reflector.go:307] Listing and watching *v1alpha1.Location from k8s.io/client-go@v0.0.0-20230210192259-aaa28aa88b2d/tools/cache/reflector.go:215
I0301 17:27:28.931078   87207 reflector.go:271] Starting reflector *v1alpha1.EdgePlacement (10h0m0s) from k8s.io/client-go@v0.0.0-20230210192259-aaa28aa88b2d/tools/cache/reflector.go:215
I0301 17:27:28.931163   87207 reflector.go:307] Listing and watching *v1alpha1.EdgePlacement from k8s.io/client-go@v0.0.0-20230210192259-aaa28aa88b2d/tools/cache/reflector.go:215
I0301 17:27:29.032037   87207 shared_informer.go:312] caches populated
I0301 17:27:29.032094   87207 shared_informer.go:312] caches populated
I0301 17:27:29.032118   87207 controller.go:159] "Starting controller" reconciler="edge-scheduler"
I0301 17:27:48.057831   87207 controller.go:132] "queueing EdgePlacement" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1"
I0301 17:27:48.057875   87207 controller.go:184] "processing key" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1"
reconciling EdgePlacement test-1 in Workspace kvdk2spgmbix
I0301 17:27:48.117386   87207 controller.go:226] "creating SinglePlacementSlice" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1" edgeplacement.workspace="kvdk2spgmbix" edgeplacement.namespace="" edgeplacement.name="test-1" edgeplacement.apiVersion="" workspace="kvdk2spgmbix" name="test-1"
I0301 17:27:48.122142   87207 controller.go:245] "created SinglePlacementSlice" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1" edgeplacement.workspace="kvdk2spgmbix" edgeplacement.namespace="" edgeplacement.name="test-1" edgeplacement.apiVersion="" workspace="kvdk2spgmbix" name="test-1"
I0301 17:28:15.123698   87207 controller.go:132] "queueing EdgePlacement" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1"
I0301 17:28:15.123747   87207 controller.go:184] "processing key" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1"
I0301 17:28:15.123772   87207 controller.go:210] "object deleted before handled" reconciler="edge-scheduler" key="kvdk2spgmbix|test-1"
received SIGINT, stopping process (will not forward signal)
```

Fixes #164

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,6 +15,7 @@ require (
 	k8s.io/code-generator v0.24.3
 	k8s.io/component-base v0.24.3
 	k8s.io/klog/v2 v2.70.1
+	k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42
```
Comment (MikeSpreitzer) [inline go.mod:18]: When I apply `go mod tidy` to this commit, it undoes the change wrt kube-openapi.  That is, `go mod tidy` puts it back the way @mra-ruiz had it.

**Review by MikeSpreitzer — COMMENTED** [review]
Oh, sorry, I forgot about those replace directives.  Good catch!

File: go.mod (hunk 1 above)
Comment (waltforme) [reply go.mod:18]: hmm, I actually tested `go mod tidy` yesterday, before submitting this PR to make sure that the command doesn't introduce changes.

Just now I tried another thing. I manually add `//indirect` (which is the only difference I see) for `k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42`, then a `go mod tidy` still insists it is 'direct' by removing that comment.

What could cause the different behaviors? My `go version` shows `go version go1.19.4 darwin/amd64`.

Comment (waltforme) [reply go.mod:18]: Aha, I found the cause. I have some extra code which is not added to git yet. That part of code directly depends on the module. Let me adjust the PR.

**Review by MikeSpreitzer — APPROVED** [review]
/LGTM

**Comment by waltforme** [comment]
Force-push to https://github.com/kcp-dev/edge-mc/commit/bb605228b2d214c07779209b154b9a1f4def1379 makes dependency on `k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42` to be indirect.

**Pull request merged**